profile.md

```diff
@@ -214,13 +214,25 @@ The following properties are defined for usage within a Kubernetes deployment:
   "namespace": "testing"
   ```
 
+- `clusterID`<br>
+  The unique identifier for the Kubernetes cluster from which the request
+  was sent. This property MUST be a non-empty string serialized as follows:
+
+  ```
+  "clusterID": "id-goes-here"
+  ```
+  For example:
+  ```
+  "clusterID": "644e1dd7-2a7f-18fb-b8ed-ed78c3f92c2b"
+  ```
+
 The following table specifies which properties MUST appear in each API:
 
 | Request API | Properties |
 | --- | --- |
-| `PUT /v2/service_instances/:instance_id` | `namespace` |
-| `PATCH /v2/service_instances/:instance_id` | `namespace` |
-| `PUT /v2/service_instances/:instance_id/service_bindings/:binding_id` | `namespace` |
+| `PUT /v2/service_instances/:instance_id` | `namespace`, `clusterID` |
```

Comment:
does MUST break the spec for backwards compatibility?

if a 2.new broker client sends this to a 2.old broker, it would ignore it and perhaps this is ok.
if a 2.old broker client tries to talk to a 2.new broker, it would error. 

That seems like a breaking change and should move to v3.

Reply:
I think we're in sort of a catch-22 situation here. We want this is be required for kube once morgan's svc-cat PR is merged.
Perhaps some wording for each field to indicate which release it was added and to then say its required for all releases after that version?

Reply:
But that is an API breaking change, and if you are using semver, the proper thing to do is to bump the major version.

Reply:
Worded correctly it wouldn't be a breaking change. If we said "starting with v2.14 all Kube platforms must include it" then all existing kubes are ok. We just need to make sure we get the clusterID PR merged into svc-cat to align with the v2.14 version bump. Old brokers are ok since they'll work with either.

Reply:
I think we have a disagreement about (semver)[https://semver.org/]. There is no wording that would make this a non-api breaking change without either a) bumping the spec to v3, or b) make it optional for all v2 implementations.

Reply:
we're in kind of a weird spot. The profile itself is optional, but we need a way for a broker to know what fields to expect AND need to be able to add new fields w/o a version bump. Note, again, adding new fields won't break brokers so it all a matter of whether the platforms are ok with a version gated new field.  Let me work on some tweaks...

Reply:
@n3wscott see what you think. While this is still technically a breaking change I'm now considering the old stuff to be a bug. I don't believe we ever intended for Kube to only get one shot at getting the list of context fields right w/o forcing a major version bump. So my new version tries to address that by at least not breaking existing Platforms but allows for us to place requirements on newer ones (all within the scope of the entire `context` thing being optional overall).

Reply:
Thanks for the update. I think this makes it more clear. And I am less un-easy about this change given it is for the profile for k8s and we have a pretty tight loop over k8s integrations. There is now a pretty clear expectation around version of the spec and how it relates to the profile.